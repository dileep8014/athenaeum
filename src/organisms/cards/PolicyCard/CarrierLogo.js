import React from 'react';
import PropTypes from 'prop-types';
import { Image } from 'cloudinary-react';
import styles from './policy_card.module.scss';

export const CarrierLogo = ({ carrierLogo }) =>
  <div className={styles['carrier-logo']}>
    <Image
      cloudName='policygenius'
      style={{ width: '105px' }}
      {...carrierLogo}
    />
    <div className={styles['divider']} />
  </div>
;

CarrierLogo.propTypes = {
  carrierLogo: PropTypes.object
};

export default CarrierLogo;
