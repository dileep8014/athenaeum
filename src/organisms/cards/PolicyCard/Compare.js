import React from 'react';
import PropTypes from 'prop-types';
import Text from 'atoms/Text';
import Hide from 'wrappers/Hide';
import Spacer from 'atoms/Spacer';
import CheckBoxField from 'molecules/formfields/CheckBoxField';
import styles from './policy_card.module.scss';

const Compare = ({ compareSelected, onCompare }) =>
  <Hide hideOn='small' className={styles['compare']}>
    <Text
      color='neutral-3'
      light
      className={compareSelected && styles['checked']}
    >
      Compare
    </Text>
    <Spacer spacer={1} />
    <CheckBoxField
      input={{
        onChange: onCompare,
        value: compareSelected
      }}
    />
    <div className={styles['divider']} />
  </Hide>

;

Compare.propTypes = {
  compareSelected: PropTypes.bool,
  onCompare: PropTypes.func
};

export default Compare;
