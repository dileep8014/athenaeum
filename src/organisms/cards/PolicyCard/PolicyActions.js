import React from 'react';
import PropTypes from 'prop-types';
import accounting from 'accounting';
import classnames from 'classnames';
import Text from 'atoms/Text';
import Layout from 'atoms/Layout';
import Button from 'atoms/Button';
import Hide from 'wrappers/Hide';
import Spacer from 'atoms/Spacer';
import styles from './policy_card.module.scss';

const ResponsiveText = ({ text, offset, size }) =>
  <svg width='100%' height='100%' viewBox={`${offset} 0 ${size} 15`}>
    <text x='0' y='12'>{text}</text>
  </svg>

;

export const PolicyActions = (props) => {
  const {
    monthlyPremium,
    discount,
    onContinue,
    onDetails,
    onCompare
  } = props;

  const formattedPremium = accounting.formatMoney(monthlyPremium);

  return (
    <div className={styles['actions']}>
      <Text type={5} color='neutral-2' semibold>
        <Text type={3} semibold>{formattedPremium}</Text>
        { ' ' }
        mo
      </Text>
      {discount && discount}
      <Spacer spacer={3} />
      <Layout mediumCols={[ 6 ]} smallCols={[ 12, 6, 6 ]}>
        <Button
          variant='action'
          onClick={onContinue}
          className={classnames(styles['continue'])}
        >
          <Hide hideOn='small'>
            <ResponsiveText text='Continue' offset={-3} size={75} />
          </Hide>
          <Hide hideOn='medium large xLarge xxLarge'>
            Continue
          </Hide>
        </Button>
        <Button onClick={onDetails}>
          <Hide hideOn='small'>
            <ResponsiveText text='Details' offset={-10} size={70} />
          </Hide>
          <Hide hideOn='medium large xLarge xxLarge'>
            Details
          </Hide>
        </Button>
        <Hide hideOn='medium large xLarge xxLarge'>
          <Button
            onClick={onCompare}
          >
            Compare
          </Button>
        </Hide>
      </Layout>
    </div>
  );
};

PolicyActions.propTypes = {
  monthlyPremium: PropTypes.number,
  discount: PropTypes.node,
  onContinue: PropTypes.func,
  onDetails: PropTypes.func,
  onCompare: PropTypes.func,
};

ResponsiveText.propTypes = {
  text: PropTypes.string.isRequired,
  offset: PropTypes.number,
  size: PropTypes.number
};

export default PolicyActions;
