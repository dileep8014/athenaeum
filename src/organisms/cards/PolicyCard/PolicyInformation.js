import React from 'react';
import PropTypes from 'prop-types';
import Hide from 'wrappers/Hide';
import Text from 'atoms/Text';
import Tooltip from 'atoms/Tooltip';
import formatValue from 'utils/formatValue';

import styles from './policy_card.module.scss';

export const PolicyInformation = ({ information }) =>
  <Hide hideOn='mobile tablet' className={styles['policy-info']}>
    {
      information.map((item, idx) =>
        <Tooltip
          text={
            <Text type={7} color='neutral-3' semibold>
              <div className={styles['policy-info-label']}>{item.label}</div>
              <Text type={7} color='neutral-2' semibold>{formatValue(item.value)}</Text>
            </Text>
          }
          hoverMessageClassName={styles['policy-type-hover-message']}
          className={styles['policy-info-tooltip']}
          key={`${item.label}-${idx}`}
        >
          { item.hoverMessage }
        </Tooltip>
      )
    }
  </Hide>
;

PolicyInformation.propTypes = {
  information: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]).isRequired,
    hoverMessage: PropTypes.node.isRequired
  })),
};

export default PolicyInformation;
