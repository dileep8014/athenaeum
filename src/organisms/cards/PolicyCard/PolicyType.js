import React from 'react';
import PropTypes from 'prop-types';
import Text from 'atoms/Text';
import Hide from 'wrappers/Hide';
import Tooltip from 'atoms/Tooltip';
import styles from './policy_card.module.scss';

export const PolicyType = ({ policyType, policyTooltip, policyTypeHoverMessage }) =>
  <div
    className={styles['policy-type']}
  >
    <Tooltip
      text={
        <div>
          <Hide hideOn='small'>
            <Text
              type={7}
              color='neutral-3'
              semibold
            >
              Policy Type
            </Text>
          </Hide>
          <Text
            color='neutral-2'
            type={5}
            semibold
            className={styles['policy-name']}
          >
            {policyType}
          </Text>
          <Text
            type={7}
            color='neutral-2'
            semibold
            className={styles['policy-tooltip']}
          >
            {policyTooltip}
          </Text>
        </div>
      }
      hoverMessageClassName={styles['policy-type-tooltip']}
    >
      { policyTypeHoverMessage }
    </Tooltip>
    <div className={styles['divider']} />
  </div>
;

PolicyType.propTypes = {
  policyType: PropTypes.string,
  policyTooltip: PropTypes.node,
  policyTypeHoverMessage: PropTypes.node,
};

export default PolicyType;
