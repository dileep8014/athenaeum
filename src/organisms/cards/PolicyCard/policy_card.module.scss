$border: #dfdfdf;
$footer: #f6f6f6;

@mixin section($grow, $centered) {
  flex: #{$grow} 1 0;
  padding: 0 ru(1.25);
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;

  @if $centered {
    align-items: center;
  }
}

.policy-card {
  border: 1px solid $border;
  border-radius: ru(.25);
  max-width: rem-calc(1200px);
}

.body {
  padding: rem-calc(20px) 0 ru(1);
  display: flex;
}

.footer {
  background-color: $footer;
  border-top: 1px solid $border;
  box-shadow: inset 0 0 10px rgba(0, 0, 0, .06);
  padding: ru(.5) ru(1.25);
  display: flex;
  flex-wrap: wrap;
}

.divider {
  height: ru(3.25);
  border-left: 1px solid $border;
  position: absolute;
  right: 0;
}

.node {
  display: inline-block;
  @include typography(7);
  color: color('neutral-3');
  padding-right: ru(.5);
}

.policy-type > strong {
  display: block;
}

.checkbox-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  cursor: pointer;
}

.checked {
  color: color('neutral-2');
  font-weight: 600;
}

.carrier-logo {
  img {
    width: 105px;
    max-width: 105px;
  }
}

.policy-tooltip {
  > * {
    display: flex;
    align-items: center;
  }
}

@media #{$small-only} {
  .policy-tooltip {
    > * {
      justify-content: center;
    }
  }

  .body {
    flex-direction: column;
  }

  .footer {
    justify-content: center;
    text-align: center;
  }

  .divider {
    display: none;
  }

  .actions, {
    padding: ru(1) ru(1) 0 ru(1);
    text-align: center;
    border-top: 1px solid $border;
  }

  .policy-type {
    padding: ru(1);
    text-align: center;
  }

  .carrier-logo {
    text-align: center;
  }

  .policy-name {
    max-width: 140px;
    margin: auto;
    margin-bottom: ru(.5);
    line-height: ru(1);
  }

  .node {
    &:first-child {
      padding-right: ru(.25);
    }
  }
}

@media #{$small-and-medium-only} {
  .continue {
    margin-bottom: ru(.75);
  }
}

@media #{$medium-up} {
  .compare {
    @include section(0, true);
    label { margin-left: rem-calc(8px); }
    min-width: ru(5.25);
  }

  .carrier-logo {
    @include section(0, true);
  }

  .policy-type {
    @include section(1, false);
  }

  .policy-info {
    @include section(1, false);
  }

  .policy-info > strong {
    padding-bottom: ru(.25);
    .label {
      display: inline-block;
      width: ru(5);
    }
  }

  .actions {
    @include section(1, true);
    text-align: center;
  }

  .policy-name {
    margin: rem-calc(3px) 0;
    line-height: ru(1);
  }
}
