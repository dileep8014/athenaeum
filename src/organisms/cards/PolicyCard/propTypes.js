import PropTypes from 'prop-types';

export default {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  /**
   * The insurance carrier logo node
   */
  carrierLogo: PropTypes.node.isRequired,
  /**
   * The type of policy
   */
  policyType: PropTypes.string.isRequired,
  /**
   * More information about the policy
   */
  policyTooltip: PropTypes.node.isRequired,
  /**
   * Details about the policy that show on hover
   */
  policyTypeHoverMessage: PropTypes.node.isRequired,
  /**
   * The financial strength of the carrier
   */
  financialStrength: PropTypes.shape({
    value: PropTypes.string.isRequired,
    hoverMessage: PropTypes.node.isRequired
  }),
  /**
   * The customer service rating of the carrier
   */
  customerService: PropTypes.shape({
    value: PropTypes.string.isRequired,
    hoverMessage: PropTypes.node.isRequired
  }),
  /**
   * The total number of customers for the carrier
   */
  totalCustomers: PropTypes.shape({
    value: PropTypes.number.isRequired,
    hoverMessage: PropTypes.node.isRequired
  }),
  /**
   * The insurance premium
   */
  premium: PropTypes.shape({
    price: PropTypes.number.isRequired,
    format: PropTypes.string.isRequired,
    defaultText: PropTypes.string,
  }),
  /**
   * Optional discount percentage
   */
  discount: PropTypes.node,
  /**
   * Optional array of nodes that will be spaced appropriately in the footer
   */
  footer: PropTypes.array,
  /**
   * Click handler for continue button
   */
  onContinue: PropTypes.func.isRequired,
  /**
   * Click handler for details button
   */
  onDetails: PropTypes.func.isRequired,
  /**
   * Props for the Checkbox to compare
   *
   * NOTE: `name` must be unique
   */
  compareCheckbox: PropTypes.shape({
    onCompare: PropTypes.func.isRequired,
    compareSelected: PropTypes.bool,
    name: PropTypes.string,
  })
};
