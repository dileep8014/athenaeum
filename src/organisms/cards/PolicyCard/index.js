import React from 'react';
import classnames from 'classnames';
import styles from './policy_card.module.scss';
import propTypes from './propTypes';

import Compare from './Compare';
import CarrierLogo from './CarrierLogo';
import PolicyType from './PolicyType';
import PolicyInformation from './PolicyInformation';
import PolicyActions from './PolicyActions';

function PolicyCard(props) {
  const {
    financialStrength,
    customerService,
    totalCustomers,
    policyType,
    policyTooltip,
    policyTypeHoverMessage,
    carrierLogo,
    premium,
    discount,
    onContinue,
    onDetails,
    compareCheckbox,
    footer,
    className,
  } = props;

  return (
    <div className={classnames(styles['policy-card'], className)}>
      <div className={styles['body']}>
        <Compare
          {...compareCheckbox}
        />
        <CarrierLogo
          carrierLogo={carrierLogo}
        />
        <PolicyType
          policyType={policyType}
          policyTooltip={policyTooltip}
          policyTypeHoverMessage={policyTypeHoverMessage}
        />
        <PolicyInformation
          financialStrength={financialStrength}
          customerService={customerService}
          totalCustomers={totalCustomers}
        />
        <PolicyActions
          onContinue={onContinue}
          onDetails={onDetails}
          onCompare={compareCheckbox.onCompare}
          discount={discount}
          premium={premium}
        />
      </div>
      {!!footer.length &&
        <div className={styles['footer']}>
          {footer.map((node, idx) =>
            <div className={styles['node']} key={`footer-${idx}`}>
              {node}
            </div>
          )}
        </div>
      }
    </div>
  );
}

PolicyCard.propTypes = propTypes;

export default PolicyCard;
