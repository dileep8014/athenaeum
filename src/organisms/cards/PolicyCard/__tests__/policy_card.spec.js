import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import PolicyCard from 'organisms/cards/PolicyCard';

describe('<PolicyCard />', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      footer: [],
      carrierLogo: <div></div>,
      policyType: 'Type',
      policyTooltip: 'Tooltip',
      financialStrength: 'A+',
      customerService: 'A+',
      totalCustomers: 5200000,
      premium: {
        price: 19.8,
        format: 'mo'
      },
      onContinue: () => {},
      onDetails: () => {},
      onCompare: () => {},
    };
  });

  it('renders correctly', () => {
    const wrapper = mount(<PolicyCard {...defaultProps} />);

    expect(wrapper.type()).toEqual(PolicyCard);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<PolicyCard {...defaultProps} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
