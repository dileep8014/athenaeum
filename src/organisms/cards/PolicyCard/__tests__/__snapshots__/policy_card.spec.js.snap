// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PolicyCard /> renders correctly 1`] = `
<div>
  <div
    className="mobile-card"
  >
    <div
      className="featured-policy-card"
    >
      <div
        className="content"
      >
        <div
          className="premium"
        >
          <strong
            className="bold neutral-1 type-b-2-medium inherit"
          >
            $19.80
             
            <strong
              className="bold neutral-2 type-b-7-medium inherit"
            >
              /MO
            </strong>
          </strong>
          <div
            className="top-1"
          />
        </div>
        <div
          className="top-6"
        />
        <div
          className="carrier-logo"
        >
          <div />
        </div>
        <div
          className="top-6"
        />
        <div
          className="policy-info"
        >
          <div>
            <div
              className="layout policy-info-item"
              id={undefined}
              style={undefined}
            >
              <div
                className="col col-sm-4 col-md-5 fullwidth border-neutral-6 policy-info-tooltip-group"
                onClick={undefined}
                style={undefined}
              >
                <p
                  className="primary-3 type-b-7-medium inherit"
                >
                  label
                </p>
                <span
                  className="tooltip-wrapper policy-info-tooltip-icon"
                  onClick={[Function]}
                >
                  <span
                    className="icon-wrapper tooltip"
                    onClick={undefined}
                    style={
                      Object {
                        "height": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <span
                    className="hover-message"
                  >
                    <div />
                  </span>
                </span>
              </div>
              <div
                className="col col-sm-2 col-md-1 border-neutral-6"
                onClick={undefined}
                style={undefined}
              >
                <div />
              </div>
              <div
                className="col col-sm-6 col-md-6 border-neutral-6"
                onClick={undefined}
                style={undefined}
              >
                <strong
                  className="bold primary-3 type-b-6-medium inherit"
                >
                  Some policy
                </strong>
              </div>
            </div>
            <div
              className="top-2"
            />
          </div>
          <div>
            <div
              className="layout policy-info-item"
              id={undefined}
              style={undefined}
            >
              <div
                className="col col-sm-4 col-md-5 fullwidth border-neutral-6 policy-info-tooltip-group"
                onClick={undefined}
                style={undefined}
              >
                <p
                  className="primary-3 type-b-7-medium inherit"
                >
                  label
                </p>
                <span
                  className="tooltip-wrapper policy-info-tooltip-icon"
                  onClick={[Function]}
                >
                  <span
                    className="icon-wrapper tooltip"
                    onClick={undefined}
                    style={
                      Object {
                        "height": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <span
                    className="hover-message"
                  >
                    <div />
                  </span>
                </span>
              </div>
              <div
                className="col col-sm-2 col-md-1 border-neutral-6"
                onClick={undefined}
                style={undefined}
              >
                <div />
              </div>
              <div
                className="col col-sm-6 col-md-6 border-neutral-6"
                onClick={undefined}
                style={undefined}
              >
                <strong
                  className="bold primary-3 type-b-6-medium inherit"
                >
                  A+
                </strong>
              </div>
            </div>
            <div
              className="top-2"
            />
          </div>
          <div
            className="top-4"
          />
        </div>
        <div
          className="button-group"
        >
          <button
            className="button"
            disabled={undefined}
            onClick={[Function]}
            type="button"
          >
            View Policy
          </button>
          <div>
            <div
              className="top-2"
            />
            <button
              className="button"
              disabled={undefined}
              onClick={[Function]}
              type="button"
            >
              Compare
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="policy-card"
  >
    <div
      className="body"
    >
      <div
        className="hide compare mobile"
      >
        <div
          className="checkbox-wrapper"
          id="undefined-checkbox-wrapper"
          onClick={[Function]}
        >
          <p
            className="neutral-3 type-b-7-medium inherit"
          >
            Compare
          </p>
          <div
            className="top-1"
          />
          <label
            className="checkbox"
            htmlFor="checkbox-undefined"
          >
            <input
              checked={undefined}
              className="checkbox-input"
              id="checkbox-undefined"
              name={undefined}
              type="checkbox"
              value={undefined}
            />
            <span
              className="checkbox-label"
            />
          </label>
        </div>
      </div>
      <div
        className="divider"
      />
      <div
        className="carrier-logo"
      >
        <div />
      </div>
      <div
        className="divider"
      />
      <div
        className="policy-type"
      >
        <span
          className="tooltip-wrapper policy-type-tooltip"
          onClick={[Function]}
        >
          <div>
            <div
              className="hide mobile"
            >
              <strong
                className="bold neutral-3 type-b-7-medium inherit"
              >
                label
              </strong>
            </div>
            <strong
              className="bold neutral-2 policy-name type-b-5-medium inherit"
            >
              Some policy
            </strong>
            <strong
              className="bold neutral-2 policy-tooltip type-b-7-medium inherit"
            >
              <div />
            </strong>
          </div>
          <span
            className="hover-message policy-type-hover-message"
          >
            <div />
          </span>
        </span>
      </div>
      <div
        className="divider"
      />
      <div
        className="hide policy-info mobile tablet"
      >
        <span
          className="tooltip-wrapper policy-info-tooltip"
          onClick={[Function]}
        >
          <strong
            className="bold neutral-3 type-b-7-medium inherit"
          >
            <div
              className="policy-info-label"
            >
              label
            </div>
            <strong
              className="bold neutral-2 type-b-7-medium inherit"
            >
              A+
            </strong>
          </strong>
          <span
            className="hover-message policy-type-hover-message"
          >
            <div />
          </span>
        </span>
        <span
          className="tooltip-wrapper policy-info-tooltip"
          onClick={[Function]}
        >
          <strong
            className="bold neutral-3 type-b-7-medium inherit"
          >
            <div
              className="policy-info-label"
            >
              label
            </div>
            <strong
              className="bold neutral-2 type-b-7-medium inherit"
            >
              A+
            </strong>
          </strong>
          <span
            className="hover-message policy-type-hover-message"
          >
            <div />
          </span>
        </span>
        <span
          className="tooltip-wrapper policy-info-tooltip"
          onClick={[Function]}
        >
          <strong
            className="bold neutral-3 type-b-7-medium inherit"
          >
            <div
              className="policy-info-label"
            >
              label
            </div>
            <strong
              className="bold neutral-2 type-b-7-medium inherit"
            >
              16
            </strong>
          </strong>
          <span
            className="hover-message policy-type-hover-message"
          >
            <div />
          </span>
        </span>
      </div>
      <div
        className="divider info-divider"
      />
      <div
        className="actions"
      >
        <div>
          <strong
            className="bold neutral-2 type-b-5-medium inherit"
          >
            <strong
              className="bold primary-3 type-b-3-medium inherit"
            >
              $19.80
            </strong>
             
            mo
          </strong>
        </div>
        <div
          className="top-3"
        />
        <div
          className="layout"
          id={undefined}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="col col-sm-12 border-neutral-6"
            onClick={undefined}
            style={undefined}
          >
            <button
              className="button"
              disabled={undefined}
              onClick={[Function]}
              type="button"
            >
              View Policy
            </button>
          </div>
          <div
            className="col col-sm-12 border-neutral-6"
            onClick={undefined}
            style={undefined}
          >
            <div
              className="hide tablet desktop"
            >
              <button
                className="button"
                disabled={undefined}
                onClick={[Function]}
                type="button"
              >
                Compare
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
